### PR TITLE
[ha-agent] Consume RC HA_AGENT

### PR DESCRIPTION
**Pull request by AlexandreYang**
### What does this PR do?

[ha-agent] Consume RC HA_AGENT

### Motivation

Need for HA Agent feature.

### Describe how to test/QA your changes

1/ Run Agent with HA Agent feature enabled:
```
ha_agent:
  enabled: true
  group: my-group01
```

2/ Write messages from backend to HA_AGENT RC product

3/ Check in logs that the RC listener for HA_AGENT is receiving the msg

### CODE REVIEW

File: comp/haagent/impl/haagent.go [hunk 1]
```diff
@@ -47,3 +49,29 @@ func (h *haAgentImpl) SetLeader(leaderAgentHostname string) {
...
+		if err != nil {
+			h.log.Warnf("Skipping invalid HA_AGENT update %s: %v", configPath, err)
+			applyStateCallback(configPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: "error unmarshalling payload",
+			})
+			continue
+		}
+
+		h.SetLeader(haAgentMsg.Leader)
```
[the remark this answers is not in the document]
Comment (AlexandreYang) [reply comp/haagent/impl/haagent.go:69]: Yeah, sure, good idea.

[the remark this answers is not in the document]
Comment (AlexandreYang) [reply comp/haagent/impl/haagent.go:78]: @mellon85 fixed here https://github.com/DataDog/datadog-agent/pull/31172/commits/73c31443491a5a1c06d63337c0dbcb88ff30fdf3

**Review by jedupau — APPROVED** [review]

File: comp/haagent/impl/haagent.go [hunk 2]
```diff
@@ -47,3 +49,38 @@ func (h *haAgentImpl) SetLeader(leaderAgentHostname string) {
...
+
+		h.log.Debugf("Processed config %s: %v", configPath, haAgentMsg)
+
+		applyStateCallback(configPath, state.ApplyStatus{
+			State: state.ApplyStateUnacknowledged,
```
[the remark this answers is not in the document]
Comment (AlexandreYang) [reply comp/haagent/impl/haagent.go]: fixed in 91ee48a641

**Review by mellon85 — APPROVED** [review]

**Pull request merged**